scripts/debug_tools/reduce_repro_files.py

```diff
@@ -0,0 +1,439 @@
+#!/usr/bin/env python
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+import argparse
+import re
+import os
+import shutil
+import stat
+import json
+import contextlib
+import io
+import sys
+import subprocess
+import multiprocessing
+import prepare_all_cmd_for_ctu
+
+
+# Suppressing output of functions
+class DummyFile(object):
+    def write(self, x): pass
+
+
+@contextlib.contextmanager
+def nostdout():
+    save_stdout = sys.stdout
+    sys.stdout = DummyFile()
+    yield
+    sys.stdout = save_stdout
+
+
+def get_analyzed_file_path(analyzer_command_file):
+    with open(analyzer_command_file, 'r') as f:
+        return f.read().split(" ")[-1]
+
+
+def get_std_flag(str):
+    std_flag_pattern = re.compile("-std=")
+    l = str.split()
+    for x in l:
+        if std_flag_pattern.match(x):
+            return x
+    return ""
+
+
+def create_ctu_dir(compile_cmds_file):
+    prepare_all_cmd_for_ctu.execute(["CodeChecker", "analyze", "--ctu-collect",
+                                     compile_cmds_file, "-o", "cc_files"],
+                                    False)
+
+
+def isCppOrCFile(file_name):
+    c_pattern = re.compile('\.c$|\.cpp$|\.cxx$')
+    return c_pattern.search(file_name)
+
+
+def get_assertion_string(analyzer_command_file):
+    error = subprocess.Popen(["bash", analyzer_command_file],
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    _, error_string = error.communicate()
+    assert_pattern = re.compile('Assertion.+failed\.')
+    assert_match = assert_pattern.search(error_string)
+    if not assert_match:
+        return ""
+    return assert_match.group(0)
+
+
+def get_compilable_cond(clang, stdflag, file_name):
+    return [clang, '-c', '-Werror', stdflag, file_name]
+
+
+def get_ast_dump_cond(clang, stdflag, file_abs_path, file_name, triple_arch):
+    ast_dump_path = os.path.join(os.path.abspath('./cc_files'),
```

Comment:
I think here you should use the `os.path.join` too: `os.path.join('.', 'cc_files')`. And cannot we use the `os.getcwd()` method to get the current working directory instead of `.`?